core/src/Cuda/Kokkos_Cuda_Parallel.hpp

```diff
@@ -1228,6 +1228,7 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
 
     *m_result_ptr =
         sum;  // is m_result_ptr always the type of pointer to value_type?
```

Comment:
```suggestion
        sum;  // is m_result_ptr always the type of pointer to value_type?
```
It might be a `nullptr` if no reducer argument was provided, i.e. the reduction is done entirely through `join` calls. For array reductions, multiple values would need to be copied.

Reply:
Given #4208, it might make sense to simply add 'array reduction' to the list of conditions that exclude the use of Thrust. I don't know if you're also considering the `View`-wrapped array case as an 'array reduction', though.

Reply:
Thanks for the comment, I have been considering the View-wrapped array case but am having a hard time getting Thrust to handle it, so I may focus on that later and for now add the "array reduction" as an exclusion as suggested